url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/logging/Generic_Log.java

```diff
@@ -0,0 +1,411 @@
+//        this.loggerFileFileHandler_HashMap = a_Generic_Log.loggerFileFileHandler_HashMap;
+//        //this._FileHandler = a_Generic_Log._FileHandler;
+//    }
+//
+//    private void init() {
+//        LogManager.getLogManager().addLogger(Logger.getLogger(Generic_DefaultLoggerName));
+//        loggerNameLevel_HashMap = new HashMap<String, Level>();
+//        loggerNameLevel_HashMap.put(Generic_DefaultLoggerName, Generic_DefaultLogLevel);
+//        loggerNameFile_HashMap = new HashMap<String, File>();
+//        loggerFileFileHandler_HashMap = new HashMap<File, FileHandler>();
+//        loggerFileHandlerFile_HashMap = new HashMap<FileHandler, File>();
+//    }
+//
+//    public Generic_Log(
+//            Level level,
+//            File directory,
+//            String classname,
+//            String filename) {
+//        init(level,
+//                directory,
+//                classname,
+//                filename);
+//    }
+//
+//    public Generic_Log(
+//            Level level,
+//            File directory,
+//            String filename) {
+//        init(level,
+//                directory,
+//                Generic_DefaultLoggerName,
+//                filename);
+//    }
+    /**
+     * Creates loggers and registers them with the LogManager. The loggers 
+     * created are configured by reading a logging.properties configuration file 
+     * found as 
+     * <code>
+     * new File(
+     * directory.toString + 
+     * System.getProperty("java.util.logging.config.file"))
+     * </code>
+     * @param loggingPropertiesFile
+     * @param directory The file directory base containing the 
+     * logging.properties file (possibly at great depth!)
+     * @param logDirectory The directory into which and LoggingFiles will be 
+     * stored (possibly at some depth)
+     * @param logname Typically the source package or some name for naming the 
+     * loggers created
+     */
+    //public static Logger parseLoggingProperties(
+    public static void parseLoggingProperties(
+            File loggingPropertiesFile,
+            File directory,
+            File logDirectory,
+            String logname) {
+        String sourceMethod = "parseLoggingProperties(File,File,File,String)";
+        //Logger logname_Logger = Logger.getLogger(logname);
+        //logger = Logger.getLogger("");
+        logger = Logger.getLogger(logname);
+        Properties logging_Properties = new Properties();
+        try {
+            FileInputStream fileInputStream = new FileInputStream(loggingPropertiesFile);
+            logging_Properties.load(fileInputStream);
+            fileInputStream.close();
+        } catch (IOException e) {
+            System.err.println(e.getMessage());
+        }
+        
+        LogManager logManager = LogManager.getLogManager();
+        try {
+            FileInputStream fileInputStream = new FileInputStream(loggingPropertiesFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fileInputStream*. The resource is referenced at line 139. The resource is closed at line 140. There are other execution paths that do not contain closure statements, for example, when an exception is thrown by LogManager.readConfiguration, FileInputStream constructor. Ensure *fileInputStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).